src/dune_rules/odoc.ml

```diff
@@ -245,14 +245,33 @@ let odoc_base_flags quiet build_dir =
   | Nonfatal -> S []
 ;;
 
+let odoc_dev_tool_lock_dir_exists () =
+  let path = Dune_pkg.Lock_dir.dev_tool_lock_dir_path Odoc in
+  Fs_memo.dir_exists (Path.source path |> Path.as_outside_build_dir_exn)
```

Comment:
Don't think the `as_outside_build_dir_exn` is necessary. You can just use of the constructors for `Path.Outside_build_dir` directly.

Reply:
Good point